clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -672,6 +672,146 @@ check_function_markers()
+        // checks do not currently work properly there.
+#if !defined(WINDOWS) || defined(X64)
+        // XXX: Just like raw2trace_unit_tests, we need to create a syscall instruction
+        // and it turns out there is no simple cross-platform way.
+#    ifdef X86
+        instr_t *sys = INSTR_CREATE_syscall(GLOBAL_DCONTEXT);
+#    elif defined(AARCHXX)
+        instr_t *sys =
+            INSTR_CREATE_svc(GLOBAL_DCONTEXT, opnd_create_immed_int((sbyte)0x0, OPSZ_1));
+#    elif defined(RISCV64)
+        instr_t *sys = INSTR_CREATE_ecall(GLOBAL_DCONTEXT);
+#    else
+#        error Unsupported architecture.
+#    endif
+        instr_t *sig_ret = XINST_CREATE_return(GLOBAL_DCONTEXT);
+
+        instr_t *label1 = INSTR_CREATE_label(GLOBAL_DCONTEXT);
+        instr_t *call1 = XINST_CREATE_call(GLOBAL_DCONTEXT, opnd_create_instr(label1));
+
+        instr_t *label2 = INSTR_CREATE_label(GLOBAL_DCONTEXT);
+        instr_t *call2 = XINST_CREATE_call(GLOBAL_DCONTEXT, opnd_create_instr(label2));
+        instr_t *ret2 = XINST_CREATE_return(GLOBAL_DCONTEXT);
+
+        instr_t *jmp = XINST_CREATE_jump(GLOBAL_DCONTEXT, opnd_create_instr(label1));
+
+        instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+        instrlist_append(ilist, call1);
+        instrlist_append(ilist, label1);
+        instrlist_append(ilist, call2);
+        instrlist_append(ilist, label2);
+        instrlist_append(ilist, ret2);
+        instrlist_append(ilist, sig_ret);
+        instrlist_append(ilist, sys);
+        instrlist_append(ilist, jmp);
+
+        static constexpr addr_t BASE_ADDR = 0x123450;
+        const int call_instr_length = instr_length(GLOBAL_DCONTEXT, call1);
+        std::vector<memref_with_IR_t> memref_setup = {
+            { gen_marker(TID, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+              nullptr },
+            { gen_instr_type(TRACE_TYPE_INSTR_DIRECT_CALL, TID), call1 },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FUNC_ID, 1), nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FUNC_RETADDR,
+                         BASE_ADDR + call_instr_length),
+              nullptr },
+
+            { gen_marker(TID, TRACE_MARKER_TYPE_KERNEL_EVENT,
+                         BASE_ADDR + call_instr_length),
+              nullptr },
+
+            { gen_instr_type(TRACE_TYPE_INSTR_DIRECT_CALL, TID), call2 },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FUNC_RETADDR,
+                         BASE_ADDR + call_instr_length * 2),
+              nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_FUNC_ID, 1), nullptr },
+
+            { gen_instr_type(TRACE_TYPE_INSTR_RETURN, TID), ret2 },
+
+            { gen_instr_type(TRACE_TYPE_INSTR_RETURN, TID), sig_ret },
+            { gen_instr(TID), sys },
+            { gen_marker(TID, TRACE_MARKER_TYPE_SYSCALL, 15), nullptr },
+            { gen_marker(TID, TRACE_MARKER_TYPE_KERNEL_XFER,
+                         BASE_ADDR + call_instr_length),
```

Comment:
This marker value looks incorrect: doesn't the checker complain about this?  It should be the syscall PC + its length.

Reply:
The checker didn't complain. I rewrote the code, and used new PCs, PTAL.